internal/worker_runner.go

```diff
@@ -62,37 +69,68 @@ func (wr *workerRunner) Run() {
 	wr.logger.Info().Msgf("found modem: %s", modem)
 
 	// we will need two clocks, one for non-obd (every 20s) and one for obd (continuous, based on each signal interval)
-	// which clock checks batteryvoltage? we want to send it with every status payload
-	// register tasks that can be iterated over
+	// battery-voltage will be checked in obd related clock to determine if it is ok to query obd
+	// battery-voltage also will be checked in non-obd clock because we want to send it with every status payload
+	go func() {
+		fingerprintDone := false
+		for {
+			// we will need to check the voltage before we query obd, and then we can query obd if voltage is ok
+			queryOBD, powerStatus := wr.isOkToQueryOBD()
+			if queryOBD {
+				fmt.Printf("voltage is enough to query obd : %f\n", powerStatus.VoltageFound)
```

Comment:
i think we can remove this, or use the `wr.logger.Debug().Msg("voltage is enough etc...")`

Reply:
updated